compiler/ilgen/IlBuilder.cpp

```diff
@@ -1443,10 +1443,53 @@ IlBuilder::Xor(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+TR::Node*
+IlBuilder::shiftOpNodeFromNodes(TR::ILOpCodes op,
+                                TR::Node *leftNode,
+                                TR::Node *rightNode) 
+   {
+   TR::DataType leftType = leftNode->getDataType();
+   TR::DataType rightType = rightNode->getDataType();
+   TR_ASSERT((leftType == TR::Int32 || leftType == TR::Int64) && rightType == TR::Int32, "shift operation first operand must be 32-bit or 64-bit integer, and shift amount must be 32-bit integer");
+
+   return TR::Node::create(op, 2, leftNode, rightNode);
+   }
+
+TR::IlValue *
+IlBuilder::shiftOpFromNodes(TR::ILOpCodes op,
+                            TR::Node *leftNode,
+                            TR::Node *rightNode) 
+   {
+   TR::Node *result = shiftOpNodeFromNodes(op, leftNode, rightNode);
+   TR::IlValue *returnValue = newValue(result->getDataType(), result);
+   return returnValue;
+   } 
+
+TR::IlValue *
+IlBuilder::shiftOpFromOpMap(OpCodeMapper mapOp,
+                            TR::IlValue *left,
+                            TR::IlValue *right)
+   {
+   if (left->getDataType() == TR::Int8 || left->getDataType() == TR::Int16)
+      left = ConvertTo(Int32, left);
+   TR_ASSERT(left->getDataType() == TR::Int32 || left->getDataType() == TR::Int64, "left operand of shift must be Int32 or Int64");
```

Comment:
What's the reason for this restriction? Unless I'm missing something, I would have thought that using one of the [`bshl`, `sshl`, `bshr`, `sshr`] opcodes would have worked here.